ext/rugged/rugged_revwalk.c

```diff
@@ -192,6 +192,35 @@ static VALUE rb_git_walker_simplify_first_parent(VALUE self)
 
 /*
  *  call-seq:
+ *    walker.count -> Fixnum
+ *
+ *  Returns the amount of objects a walker iterated over.
+ */
+static VALUE rb_git_walker_count(VALUE self)
+{
+    git_revwalk *walk;
+    git_oid commit_oid;
+    int error = 0;
+    long count = 0;
+    uint64_t limit = UINT64_MAX;
+
+    Data_Get_Struct(self, git_revwalk, walk);
+
+    while ((error = git_revwalk_next(&commit_oid, walk)) == 0) {
```

Comment:
I believe if you do something like:

```
while ((error = git_revwalk_next(&commit_oid, walk)) == 0) && ++count != UINT64_MAX);
```

you could remove the limit variable altogether.